Android/app/src/main/java/com/example/boardgamesocial/SignUpActivity.java

```diff
@@ -0,0 +1,44 @@
+package com.example.boardgamesocial;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.widget.Button;
+import android.widget.EditText;
+
+public class SignUpActivity extends AppCompatActivity {
+
+    private EditText etUsername, etPassword;
+    private Button btnRegister, btnSignIn;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_sign_up);
+
+        wireUpDisplay();
+    }
+
+    private void wireUpDisplay() {
+        etUsername = findViewById(R.id.et_signup_username);
+        etPassword = findViewById(R.id.et_signup_password);
+        btnRegister = findViewById(R.id.btn_register);
+        btnSignIn = findViewById(R.id.btn_back_to_login);
+
+        btnRegister.setOnClickListener(view -> {
+            //TODO: verify if user already exits!
```

Comment:
If you try to sign up with user credentials that already exist you will get 400 error from backend. So you can just send the request and check if it is 400 or 200 response status and proceed from there.